src/sentry/api/base.py

```diff
@@ -463,3 +465,23 @@ def track_set_commits_local(self, request: Request, organization_id=None, projec
             project_ids=project_ids,
             user_agent=request.META.get("HTTP_USER_AGENT", ""),
         )
+
+
+class _InactiveEndpoint(Endpoint):
+    def dispatch(self, request: Request, *args, **kwargs) -> HttpResponse:
+        return HttpResponse(status=status.HTTP_404_NOT_FOUND)
+
+
+def mode_limited(mode: ServerComponentMode) -> Callable[[type], type]:
```

Comment:
@RyanSkonnord just throwing some ideas for other names: `@active_when(ServerComponentMode.CUSTOMER)`

Reply:
Other names options that come to mind for me:

* active_for
* active_on
* enabled_for
* limited_to

Reply:
I'm liking `active_on` at the moment.